UI/CombatReport/CombatReportWnd.cpp

```diff
@@ -116,31 +185,94 @@ class CombatReportWnd::CombatReportPrivate {
         DebugLogger() << "RectangleHover " << data;
         SetFocus(data);
     }
+
+    void UpdateMinSize() {
+        m_min_size = GG::Pt(GG::X0, GG::Y0);
+
+        m_min_size += m_wnd.Size() - m_wnd.ClientSize();
+
+        // The rest of this function could use m_tabs->MinUsableSize instead of
+        // dealing with the children of m_tabs directly, but that checks the
+        // MinUsableSize of _all_ child windows, not just the currently
+        // selected one.
+        m_min_size += m_tabs->CurrentWnd()->MinUsableSize();
+
+        std::list<GG::Wnd*>::const_iterator layout_begin =
+            m_tabs->GetLayout()->Children().begin();
+        // First object in the layout should be the tab bar.
+        if(layout_begin != m_tabs->GetLayout()->Children().end()) {
+            GG::Pt tab_min_size = (*layout_begin)->MinUsableSize();
+            m_min_size.x = std::max(tab_min_size.x, m_min_size.x);
+            // TabBar::MinUsableSize does not seem to return the correct
+            // height, so extra height is added here.
+            m_min_size.y += tab_min_size.y + GG::Y(14);
+        }
+
+        // If the window is currently too small, re-validate its size.
+        if(m_wnd.Width() < m_min_size.x || m_wnd.Height() < m_min_size.y) {
+            m_wnd.Resize(m_wnd.Size());
+        }
+    }
+
+    void HandleTabChanged(int index) {
+        // Use the minimum size of the newly selected window.
+        UpdateMinSize();
+
+        // Make sure that the newly selected window gets an update.
+        DoLayout();
+    }
 };
 
 CombatReportWnd::CombatReportWnd() :
     CUIWnd(UserString("COMBAT_REPORT_TITLE"), GG::X(150), GG::Y(50), COMBAT_LOG_WIDTH, COMBAT_LOG_HEIGHT,
            GG::INTERACTIVE | GG::RESIZABLE | GG::DRAGABLE | GG::ONTOP | CLOSABLE),
-    m_impl(new CombatReportPrivate(*this))
-{ }
+    m_impl(0)
+{ m_impl.reset(new CombatReportPrivate(*this)); }
 
 CombatReportWnd::~CombatReportWnd()
-{}
+{ }
 
 void CombatReportWnd::SetLog(int log_id)
 { m_impl->SetLog(log_id); }
 
-GG::Pt CombatReportWnd::ClientLowerRight() const {
-    GG::Pt lr = CUIWnd::ClientLowerRight();
-    lr.x -= RESIZE_MARGIN_X;
-    lr.y -= RESIZE_MARGIN_Y;
-    return lr;
-}
-
 void CombatReportWnd::CloseClicked()
 { Hide(); }
 
 void CombatReportWnd::SizeMove(const GG::Pt& ul, const GG::Pt& lr) {
-    CUIWnd::SizeMove(ul, lr);
+    // Using Set(Min|Max)Size would be more elegant but that requires knowing
+    // when the app window changes size.  HumanClientApp (or its superclasses)
+    // don't send signals when switching to fullscreen mode, so we can't rely
+    // on calling SetMaxSize when that changes - we have to perform the size
+    // checks here.
+    CUIWnd::SizeMove(ul, ul + m_impl->ValidateSize(lr - ul));
     m_impl->DoLayout();
 }
+
+bool CombatReportWnd::EventFilter(GG::Wnd* w, const GG::WndEvent& wnd_event) {
```

Comment:
Rather than doing all this complicated event filtering, you could just adjust the used space for the graph / log to not go so far down that it covers the resize tab. That is what's done in many other UI windows...